### PR TITLE
Allow hash path in redirectUrl

### LINKED ISSUE #2307: Redirect url should be urlencoded
The user verify link is made in such way that it provides redirect path in the middle of whole link.

The problem is that when one would use `redirect=/#/my-angular-path` then the token is treated as hash valu and isn't read by api (to verify) and api throws error.
Redirect is added on line [407](https://github.com/strongloop/loopback/blob/6964914bab64496c789c434fec39d2914231ee23/common/models/user.js#L407) and token on [431](https://github.com/strongloop/loopback/blob/6964914bab64496c789c434fec39d2914231ee23/common/models/user.js#L431).

I would just remove:

```
   '&redirect=' +
      options.redirect;
```

from line 407 and put it behind 
`options.verifyHref += '&token=' + user.verificationToken;`
on line 431

### PR DESCRIPTION
**Pull request by ambrt**
### Description
Allows GET variables and hash paths for redirecting user after email confirm.

#### Related issues

- #2307

### CODE REVIEW

**Comment by slnode** [comment]
Can one of the admins verify this patch?

**Comment by slnode** [comment]
Can one of the admins verify this patch?

**Comment by slnode** [comment]
Can one of the admins verify this patch?

**Comment by slnode** [comment]
Can one of the admins verify this patch?

File: common/models/user.js [hunk 1]
```diff
@@ -814,7 +814,7 @@ module.exports = function(User) {
         if (!ctx.res) {
           return next(new Error(g.f('The transport does not support HTTP redirects.')));
         }
-        ctx.res.location(ctx.args.redirect);
+        ctx.res.location(decodeURIComponent(ctx.args.redirect));
```
Comment (bajtos) [inline common/models/user.js]: I believe `ctx.args.redirect` should have been already url-decoded by strong-remoting, therefore this change is not necessary. Unless I am missing something?

Comment (ambrt) [inline common/models/user.js]: If its decoded then this change is not necessary.

**Comment by bajtos** [comment]
Closing in favour of https://github.com/strongloop/loopback/pull/3185.

**Pull request closed without merging**